EX05/ProxyServer.py

```
@@ -93,24 +95,32 @@ def _connect_both_sides(self, entry):
             self._add_to_database(accepted, active, state=connection)
             self._add_to_database(active, accepted, state=connection)
         except Exception as e:
-            for s in socket_list:
-                s.close()
-                fd_socekt = s.fileno()
-                if fd_socekt in self._database.keys():
-                    self._database.pop(fd_socekt)
+            self._close_socket_error_in_creation(accepted)
+            self._close_socket_error_in_creation(active)
+
+    def _close_socket_error_in_creation(self, maybe_socket):
+        if maybe_socket is not None:
+            fd_socekt = maybe_socket.fileno()
+            maybe_socket.close()
+            if fd_socekt in self._database.keys():
+                self._database.pop(fd_socekt)
 
     def _build_poller(self):
         poller = select.poll()
         for fd in self._database:
-            entry = self._database[fd]
             events = select.POLLERR
-            if entry["buff"]:
-                events |= select.POLLOUT
-            if entry["state"]:
+            if fd["state"] == CLOSE:
+                events |= socket.POLLOUT
+            elif fd["state"] == PROXY:
+                events |= socket.POLLIN
+            else:
```

why else? please be explicit, if non matches you should raise an error as it is internal error.